Container Documentation (#687)
NOAA-GFDL/MDTF-diagnostics: doc/sphinx/ref_container.rst

.. role:: code-rst(code)
   :language: reStructuredText
.. _ref-container:
Container Reference
===============================
This section provides basic directions for downloading,
installing, and running the example_multicase POD in the
Model Diagnostics Task Force (MDTF) container.

Getting the Container
-------------------------------

> The container assumes that the MDTF GitHub repo is located on your local machine.

Reviewer: ```suggestion
The container assumes that the MDTF-diangnostics GitHub repo is located on your local machine.
```

      git clone https://github.com/NOAA-GFDL/MDTF-diagnostics.git

The container can then be pulled from the GitHub
container registry with the command:

   .. code-block:: bash

      docker pull ghcr.io/noaa-gfdl/mdtf-diagnostics:container

or with the equivalent command in your container software.
If you do not have a container software, Docker can be downloaded from `here <https://docs.docker.com/desktop/>`_.

Launching the Container
-------------------------------
The container itself can be launched with Docker using:

   .. code-block:: bash

      docker run -it -v {DIAG_DIR}:/proj/MDTF-diagnostics/diagnostics/ -v {WKDIR}:/proj/wkdir mdtf

wherein:
   * :code-rst:`{DIAG_DIR}` is the path to the diagnostics directory on your local machine.
     This volume is not required, but heavily recommended.
   * :code-rst:`{WKDIR}` is where you would like to store the output on your local machine.
     This allows the output HTML to be reachable without having to open a port to the container.

These happen to be the only required volumes. Further volumes may need to be mounted including volumes such as data storage.

Generating Synthetic Data
-------------------------------

> Now that we are in the container, we can get to making some data to run the POD with.

Reviewer: ```suggestion
Now that we are in the container, we can create some data to run the POD with.
```

The MDTF has a synthetic data generator for just this case. First, move into the MDTF-diagnostics dir:

   .. code-block:: bash

      cd /proj/MDTF-diagnostics/

> We can now use the following to generate our synthetic data:

Reviewer: ```suggestion
We generate our synthetic data by running:
```
